lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -503,6 +507,18 @@ def parse(self, inventory, loader, path, cache=True):
                             self._add_hosts(value['instances'], config_data, project_disks=project_disks)
                             cached_data[project][zone] = value['instances']
 
+                # Handle aggregated list.
+                while resp.get('nextPageToken'):
```

Comment:
To not repeat the for loop two times, maybe this while could look something like this?
```
first_resp = True
while first_respo or resp.get('nextPageToken'):
    if not first_resp:
        resp = self.fetch_list(params, link, query, resp.get('nextPageToken'))
    else:
        first_respo = False
....
```
and delete the above for loop.